internal/pkg/provision/providers/qemu/qemu.go

```diff
@@ -0,0 +1,75 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package qemu
+
+import (
+	"context"
+
+	"github.com/talos-systems/talos/internal/app/machined/pkg/runtime"
+	"github.com/talos-systems/talos/internal/pkg/provision"
+	"github.com/talos-systems/talos/internal/pkg/provision/providers/vm"
+	"github.com/talos-systems/talos/pkg/config/types/v1alpha1"
+	"github.com/talos-systems/talos/pkg/config/types/v1alpha1/generate"
+)
+
+type provisioner struct {
+	vm.Provisioner
+}
+
+// NewProvisioner initializes qemu provisioner.
+func NewProvisioner(ctx context.Context) (provision.Provisioner, error) {
+	p := &provisioner{
+		vm.Provisioner{
+			Name: "qemu",
+		},
+	}
+
+	return p, nil
+}
+
+// Close and release resources.
+func (p *provisioner) Close() error {
+	return nil
+}
+
+// GenOptions provides a list of additional config generate options.
+func (p *provisioner) GenOptions(networkReq provision.NetworkRequest) []generate.GenOption {
+	nameservers := make([]string, len(networkReq.Nameservers))
+	for i := range nameservers {
+		nameservers[i] = networkReq.Nameservers[i].String()
+	}
+
+	return []generate.GenOption{
+		generate.WithInstallDisk("/dev/vda"),
+		generate.WithInstallExtraKernelArgs([]string{
+			"console=ttyS0",
+			// reboot configuration
+			"reboot=k",
+			"panic=1",
+			// disable stuff we don't need
+			"pci=off",
+			"acpi=off",
+			"i8042.noaux=",
+			// Talos-specific
+			"talos.platform=metal",
```

Comment:
this is something we'll fix later, but with firecracker we had each node loading its own config by kernel command line args, with qemu we probably need a bit of a different way, e.g. we could deliver different config from http server based on node IP so that kernel arg is same for all the nodes (otherwise need a different config per node, which is also doable)

but that is a thought for the future

Reply:
yeah, I guess we can think about it when we get network in place.